api/include/opentelemetry/common/macros.h

```diff
@@ -175,3 +175,28 @@ point.
 #  define OPENTELEMETRY_API_SINGLETON
 
 #endif
+
+/* clang-format on */
+//
+// The if/elif order matters here. If both OPENTELEMETRY_BUILD_IMPORT_DLL and
+// OPENTELEMETRY_BUILD_EXPORT_DLL are defined, the former takes precedence.
+//
+// TODO: consider define OPENTELEMETRY_EXPORT for cygwin/gcc, see below link.
+// https://gcc.gnu.org/wiki/Visibility#How_to_use_the_new_C.2B-.2B-_visibility_support
+//
+#if defined(_MSC_VER) && defined(OPENTELEMETRY_BUILD_IMPORT_DLL)
```

Comment:
I think we can use `target_compile_definitions(opentelemetry_api INTERFACE OPENTELEMETRY_BUILD_IMPORT_DLL)` to export this definition into cmake CONFIG file, and check `OPENTELEMETRY_BUILD_EXPORT_DLL` first.And then users do not need to declare `OPENTELEMETRY_BUILD_IMPORT_DLL` themself when using cmake.

Reply:
Do you mean when CMake flag `OPENTELEMETRY_BUILD_DLL` is defined, add `OPENTELEMETRY_BUILD_IMPORT_DLL` to the `opentelelemtry_api` target to user to link? If so, then yes, the user doesn't need define `OPENTELEMETRY_BUILD_EXPORT_DLL` as C++ macro, but this only applies to CMake user. The current approach gives user API header files and .dll, then the user can referent it with any build system, CMake bazel or others. Just define a C++ macro `OPENTELEMETRY_BUILD_EXPORT_DLL` before including our header files will work.

But I think we can make some improvement for CMake user with `find_package(...)`. Can we do this in later PR?

Reply:
> Do you mean when CMake flag `OPENTELEMETRY_BUILD_DLL` is defined, add `OPENTELEMETRY_BUILD_IMPORT_DLL` to the `opentelelemtry_api` target to user to link? If so, then yes, the user doesn't need define `OPENTELEMETRY_BUILD_EXPORT_DLL` as C++ macro, but this only applies to CMake user. The current approach gives user API header files and .dll, then the user can referent it with any build system, CMake bazel or others. Just define a C++ macro `OPENTELEMETRY_BUILD_EXPORT_DLL` before including our header files will work.
> 
> But I think we can make some improvement for CMake user with `find_package(...)`. Can we do this in later PR?

Yes, that's what I mean.We can do it in another PR.